azure/ConsiderationsForServiceDesign.md

```diff
@@ -0,0 +1,99 @@
+
+Your goal is to create a developer friendly API where:
+
+> :white_check_mark: **DO** ensure that customer workloads never break
+> 
+> :white_check_mark: **DO** ensure that customers are able to adopt a new version of service or SDK client library <b>without requiring code changes</b> 
+
+### Azure Management Plane vs Data Plane
+*Note: Developing a new service requires the development of at least 1 (management plane) API and potentially one or more additional (data plane) APIs.  When reviewing v1 service APIs, we see common advice provided during the review.*
+
+A **management plane** API is implemented through the Azure Resource Manager (ARM) and is used by subscription administrators. A **data plane** API is used by developers to implement applications. Occasionally, some operations are useful to both administrators and developers. In this case, the operation can appear in both APIs. Although, best practices and patterns described in this document apply to all HTTP/REST APIs, they are especially important for **data plane** services because it is the primary interface for developers using your service. The **management plane** APIs may have other preferred practices based on [the conventions of the Azure ARM](https://github.com/Azure/azure-resource-manager-rpc).
+
+### Start with the Developer Experience
+A great API starts with a well thought out and designed service. Your service should define simple/understandable abstractions with each given a clear name that you use consistently throughout your API and documentation. There must also be an unambiguous relationship between these abstractions.
+
+It is extremely difficult to create an elegant API that works well on top of a poorly designed service; the service team and customers will live with this pain for years to come. So, the service team should empathize with customers by:
+ - Building apps that consume the API
+ - Hold reviews and share what is learned with your team
+ - Get customer feedback from API previews
+ - Thinking about the code that a customer writes both before and after an HTTP operation
+ - Initializing and reading from the data structures your service requires
+ - Thinking about which errors are recoverable at runtime as opposed to indicating a bug in the customer code that must be fixed
+
+The whole purpose of a preview to address feedback by improving abstractions, naming, relationships, API operations, and so on. It is OK to make breaking changes during a preview to improve the experience now so that it is sustainable long term.
+
+The API should be designed so that developers can use their language of choice to access the service.
+
+> :no_entry: **DO** restrict integer values to the range of +/- 2<sup>53</sup>, since JavaScript native types cannot represent integers outside this range.
```

Comment:
Same here. It sounds like just an example of many guidelines that would follow. Is that the intention?

Reply:
Agreed that this is just one of many examples and it is actually redundant with guidance we already have in the Guidelines.md so I will just remove it.